javatests/arcs/core/data/RecipeTest.kt

```diff
@@ -110,17 +110,20 @@ class RecipeTest {
             name = "ParticleName",
             location = "com.Particle",
             connections = listOf(
-                HandleConnectionSpec("data", HandleMode.Read, contactCollectionType)
+                HandleConnectionSpec("data", HandleMode.Read, personCollectionType)
             ).associateBy { it.name }
         )
 
+        // Given that the ParticleSpec type is different from the HandleConnection instance type,
+        // prefer the HandleConnection instance type for the translation.
```

Comment:
@piotrswigon I modified this test for the update. I can also add a new test, but I wanted to be a little more lofi.